common/src/main/java/io/druid/common/guava/GuavaUtils.java

```diff
@@ -52,47 +51,36 @@ public String apply(@Nullable String input)
     };
   }
 
-  public static InputSupplier<BufferedReader> joinFiles(final File... files)
+  public static CharSource joinFiles(final File... files)
   {
     return joinFiles(Arrays.asList(files));
   }
 
-  public static InputSupplier<BufferedReader> joinFiles(final List<File> files)
+  public static CharSource joinFiles(final List<File> files)
   {
-
-    return new InputSupplier<BufferedReader>()
-    {
-      @Override
-      public BufferedReader getInput() throws IOException
-      {
-        return new BufferedReader(
-            CharStreams.join(
-                Iterables.transform(
-                    files,
-                    new Function<File, InputSupplier<InputStreamReader>>()
-                    {
-                      @Override
-                      public InputSupplier<InputStreamReader> apply(final File input)
-                      {
-                        return new InputSupplier<InputStreamReader>()
-                        {
-                          @Override
-                          public InputStreamReader getInput() throws IOException
-                          {
-                            InputStream baseStream = new FileInputStream(input);
-                            if (input.getName().endsWith(".gz")) {
-                              baseStream = new GZIPInputStream(baseStream);
-                            }
-
-                            return new InputStreamReader(baseStream, Charsets.UTF_8);
-                          }
-                        };
-                      }
+    return CharSource.concat(
+        Iterables.transform(
+            files,
+            new Function<File, CharSource>()
+            {
+              @Override
+              public CharSource apply(final File input)
+              {
+                return new CharSource()
+                {
+                  @Override
+                  public Reader openStream() throws IOException
+                  {
+                    InputStream baseStream = new FileInputStream(input);
+                    if (input.getName().endsWith(".gz")) {
+                      baseStream = new GZIPInputStream(baseStream);
```

Comment:
While we're here can you make this wrapped by `com.metamx.common.CompressionUtils#gzipInputStream` ?